### PR TITLE
I created the app, which gets prime numbers from the array

### PR DESCRIPTION
**Pull request by Ilia93**

### CODE REVIEW

File: src/task_4/array/StartOfApp.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package task_4.array;
+
+import task_4.global.Variables;
+import task_4.output.OutputResult;
+
+public class StartOfApp {
```
Comment (andreikastsiuk) [inline src/task_4/array/StartOfApp.java]: StartOfApp class is optional. Actually, you can move the code to Main class.

File: src/task_4/array/StartOfApp.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public void start() {
+        OutputResult result = new OutputResult();
+        Variables variables = new Variables();
```
Comment (andreikastsiuk) [inline src/task_4/array/StartOfApp.java]: Variables should be Enum or Global constants

File: src/task_4/array/StartOfApp.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        String privateChose = variables.getChose();
+        String privateYes = variables.getYes();
+        String privateNo = variables.getNo();
+
+        result.println("Hello! You start the app for getting prime numbers. " + "Would you like to make an number array? Enter Y/N:");
+        privateChose = Variables.scanner.next();
+        if (privateChose.equals(privateYes)) {
+           ArrayInitialization getArray = new ArrayInitialization();
```
Comment (andreikastsiuk) [inline src/task_4/array/StartOfApp.java]: getArray is strange name of instance. Name have to be a noun

File: src/task_4/array/ArrayInitialization.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+public class ArrayInitialization {
+    public void array() {
+        OutputResult result = new OutputResult();
+        Variables variables = new Variables();
+        String privateChose = variables.getChose();
+        String privateYes = variables.getYes();
+        String privateNo = variables.getNo();
+
+        result.print("Your massive are: ");
+        for (int i = 2; i < Variables.array.length; i++) {
```
Comment (andreikastsiuk) [inline src/task_4/array/ArrayInitialization.java]: Variables.array.length don't use properties directly. It's a bad movetone. Also, you violate security of class state.

File: src/task_4/global/Variables.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package task_4.global;
+
+import java.util.Scanner;
+
+public final class Variables {
+    public static Scanner scanner = new Scanner(System.in);
+    public static final Integer[] array = new Integer[101];
+
+    private final String yes = "y";
```
Comment (andreikastsiuk) [inline src/task_4/global/Variables.java]: make this variables as public constants

File: src/task_4/global/Variables.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package task_4.global;
+
+import java.util.Scanner;
+
+public final class Variables {
+    public static Scanner scanner = new Scanner(System.in);
+    public static final Integer[] array = new Integer[101];
```
Comment (andreikastsiuk) [inline src/task_4/global/Variables.java]: Could you clarify why the array has got size 101?

File: src/task_4/prime_numbers_algorithm/PrimeNumbersAlgorithm.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+public class PrimeNumbersAlgorithm {
+    OutputResult result = new OutputResult();
+
+    public void primeAlgorithm() {
+        int i, j;
+        boolean isPrime;
+        for (i = 2; i < Variables.array.length; i++) {
+            isPrime = true;
+            // проверить, делится ли число без остатка
```
Comment (andreikastsiuk) [inline src/task_4/prime_numbers_algorithm/PrimeNumbersAlgorithm.java]: please, try to use only English language

File: src/task_4/prime_numbers_algorithm/PrimeNumbersAlgorithm.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+            for (j = 2; j <= i / j; j++) {
+                // если число делится без остатка, значит, оно не простое
+                if ((i % j) == 0) {
```
Comment (andreikastsiuk) [inline src/task_4/prime_numbers_algorithm/PrimeNumbersAlgorithm.java]: try to simplify the second loop ( you don't need to do two times verification)

File: src/task_4/global/Variables.java (hunk 6 above)
Comment (Ilia93) [reply src/task_4/global/Variables.java]: If I make an array of 100, then I get the numbers from 2 to 99, but not to 100, as in the condition

**Pull request merged**